Makefile

```diff
@@ -271,7 +271,11 @@ install-deps-for-semgrep-core: semgrep.opam
 	&& ./configure \
 	&& ./scripts/install-tree-sitter-lib
 	# Install OCaml dependencies (globally) from *.opam files.
-	opam install -y --deps-only ./ ./libs/ocaml-tree-sitter-core
+	# This now also installs the dev dependencies. This has the benefit
+	# of installing all the packages in one shot and detecting possible
+	# version conflicts.
+	opam install -y --deps-only \
```

Comment:
not a fan. In fact this cause conflicts in the ocaml5 build job as you can see in CI.

Reply:
The dev packages are not optional to us, developers. We need ocamlformat. The conflicts were not detected previously. It's a good thing that we now detect them.

Reply:
I'm splitting the dev dependencies between required and optional.

Reply:
So I did that: there's still a `make dev-setup` that installs optional dependencies. Installing ocamlformat is now a required dev dependency which is a new status.